### PR TITLE
docs/design: update the template of RFC docs

### PR DESCRIPTION
**Pull request by bb7133**
- add 'Testing Plan' part
- add 'Mirgration Plan' to 'Compatibility' part with descriptions

### CODE REVIEW

File: docs/design/TEMPLATE.md [hunk 1]
```diff
@@ -48,12 +48,14 @@ A discussion of alternate approaches and the trade-offs, advantages, and disadva
...
 
-## Compatibility
+## Compatibility and Mirgration Plan
 
 <!--
 A discussion of the change with regard to the compatibility issues:
 - Does this proposal make TiDB not compatible with the old versions?
+- If the existing behavior will be changed, how will we phase out the older behavior?
 - Does this proposal make TiDB more compatible with MySQL?
+- What is the impact(if any) will be on the mirgration from MySQL to TiDB?
```
Comment (zz-jason) [inline docs/design/TEMPLATE.md]: how about:

```
- what is the impact(if any) on the data migration:
	- from MySQL to TiDB
	- from TiDB to MySQL
	- from old TiDB cluster to new TiDB cluster
```

File: docs/design/TEMPLATE.md [hunk 2]
```diff
@@ -66,6 +68,14 @@ A detailed description for each step in the implementation:
 - How long it takes to accomplish it?
 -->
 
+## Testing Plan
+
+<!--
+A brief description on how the implementation will be tested. The framework tests should be focused instead of unit tests:
```
[the remark this answers is not in the document]
Comment (bb7133) [reply docs/design/TEMPLATE.md]: Addressed, thanks.

File: docs/design/TEMPLATE.md (hunk 1 above)
Comment (bb7133) [reply docs/design/TEMPLATE.md]: Addressed.

File: docs/design/TEMPLATE.md [hunk 3]
```diff
@@ -48,12 +48,17 @@ A discussion of alternate approaches and the trade-offs, advantages, and disadva
 - What is the impact of not doing this?
 -->
 
-## Compatibility
+## Compatibility and Mirgration Plan
```
Comment (Deardrops) [inline docs/design/TEMPLATE.md]: typo：
```suggestion
## Compatibility and Migration Plan
```

File: docs/design/TEMPLATE.md [hunk 4]
```diff
@@ -48,12 +48,17 @@ A discussion of alternate approaches and the trade-offs, advantages, and disadva
...
 
 <!--
 A discussion of the change with regard to the compatibility issues:
 - Does this proposal make TiDB not compatible with the old versions?
+- If the existing behavior will be changed, how will we phase out the older behavior?
 - Does this proposal make TiDB more compatible with MySQL?
+- What is the impact(if any) on the data migration:
```
Comment (Deardrops) [inline docs/design/TEMPLATE.md]:
```suggestion
- What is the impact(if any) on the data migrations:
```

**Comment by coocood** [comment]
LGTM

**Pull request merged**